Src/EmbeddedScripts.CSharp.Roslyn.Scripting/ScriptCodeRunner.cs

```diff
@@ -12,26 +12,30 @@ public class ScriptCodeRunner : ICodeRunner
     {
         private Container _container = new();
         private ScriptOptions _roslynOptions = ScriptOptions.Default;
+        private ScriptState _scriptState;
 
         public ScriptCodeRunner AddEngineOptions(Func<ScriptOptions, ScriptOptions> optionsFunc)
         {
             _roslynOptions = optionsFunc(_roslynOptions);
             return this;
         }
-        
-        public async Task RunAsync(string code)
+
+        public async Task<ICodeRunner> RunAsync(string code)
         {
             try
             {
-                await CSharpScript.RunAsync(
-                    GenerateScriptCode(code),
-                    BuildEngineOptions(),
-                    new Globals {Container = _container});
+                if (_scriptState is null)
```

Comment:
Does the latter code path (ContinueWithAsync) respect changes made to options made after the first run (eg newly registered globals)?

Reply:
Yes, it does. There is a test for newly registered globals and I've added such feature and test for options.